src/conn/conn_compact.c

```diff
@@ -308,15 +304,19 @@ __background_compact_find_next_uri(WT_SESSION_IMPL *session, WT_ITEM *uri, WT_IT
             ret = WT_NOTFOUND;
             break;
         }
-        /*
-         * Check the list of files background compact has tracked statistics for. This avoids having
-         * to open a dhandle for the file if compaction is unlikely to work efficiently on this
-         * file.
-         */
-        WT_ERR(cursor->get_value(cursor, &value));
-        WT_ERR(__wt_config_getones(session, value, "id", &id));
-        if (__background_compact_should_run(session, key, id.val))
-            break;
+
+        /* Check the file is eligible for compaction. */
+        if (__wt_compact_check_eligibility(session, key)) {
+            /*
+             * Check the list of files background compact has tracked statistics for. This avoids
+             * having to open a dhandle for the file if compaction is unlikely to work efficiently
+             * on this file.
+             */
+            WT_ERR(cursor->get_value(cursor, &value));
+            WT_ERR(__wt_config_getones(session, value, "id", &id));
+            if (__background_compact_should_run(session, key, id.val))
```

Comment:
I think you should leave this call as is and put `__wt_compact_check_eligibility(session, key)` inside `(__background_compact_should_run(session, key, id.val)`.

Reply:
I wanted to avoid the cost of the functions `get_value` and `__wt_config_getones` especially since the latter deals with string parsing.

Reply:
That's a good point!